go/v1/resource_descriptor.go

```diff
@@ -4,15 +4,33 @@ Wrapper APIs for in-toto attestation ResourceDescriptor protos.
 
 package v1
 
-import "errors"
+import (
+	"encoding/hex"
+	"errors"
+)
 
-var ErrRDRequiredField = errors.New("at least one of name, URI, or digest are required")
+var (
+	ErrInvalidDigestEncoding = errors.New("digest is not valid hex-encoded string")
+	ErrRDRequiredField       = errors.New("at least one of name, URI, or digest are required")
+)
 
 func (d *ResourceDescriptor) Validate() error {
 	// at least one of name, URI or digest are required
 	if d.GetName() == "" && d.GetUri() == "" && len(d.GetDigest()) == 0 {
 		return ErrRDRequiredField
 	}
 
+	// the in-toto spec expects a hex-encoded string in DigestSets
```

Comment:
Note that this is only expected for the named standard algorithms listed on the page: https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md#:~:text=The%20named%20standard%20algorithms.

For other algorithm types there's no restriction.

FWIW we are using some other types and their values wind up being non-hex-encoded.

Reply:
ah, interesting. So even this check should really be made conditional on the algorithm, making Aditya's suggestion to check length an easier addition, still.

Reply:
> For other algorithm types there's no restriction.

Is this by design or is it more of a loophole? :thinking:

Reply:
By design!  For example, we have some immutable IDs that start with `1-`.  We might also have some IDs which are immutable based on underlying implementation that are just digits.

We even tell folks "New algorithms MUST document how the value is encoded, e.g. URL-safe base64, lowercase hex, etc..." which indicates that new algorithms have the choice.